File: client/projects/client-app/src/app/shared/shared/models/discussion.ts
export interface Discussion {
  communityId: number;
  id: number;
  title: string;
  authorName?: string;
  authorId: string;
  votesUp: number;
  votesDown: number;
  content: number;
  createdAt: Date;
  editedAt?: Date;
  commentsCount: number;
}

File: api/Domain/Entities/Discussion.cs
using Sociussion.Domain.Common;

namespace Sociussion.Domain.Entities;

public class Discussion : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;

    public DateTime? LastActive { get; set; } = null;
    public int VotesUp { get; set; } = 0;
    public int VotesDown { get; set; } = 0;
    
    public ApplicationUser Author { get; set; }
    public int? AuthorId { get; set; }

    public Community Community { get; set; }
    public int CommunityId { get; set; }

    public ICollection<Comment> Comments { get; set; }
    public int CommentCount { get; set; } = 0;

    public ICollection<ApplicationUser> SavedBy { get; set; }
}